### PR TITLE
Feature/SLDT - adds digital twin feature

### PR DESCRIPTION
**Pull request by JaneMcBrain**
Implements the following designs:
Twin overview: https://www.figma.com/file/pcHZWXTQJUguIT75QSk9KW/Portal-Designs?node-id=1010%3A86762
Twin detail: https://www.figma.com/file/pcHZWXTQJUguIT75QSk9KW/Portal-Designs?node-id=851%3A75024
Includes also a first Accordion component.

Jira ticket: https://jira.catena-x.net/browse/A1SLDT-476

### CODE REVIEW

File: cx-portal/src/components/pages/DigitalTwins/Header.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { Typography } from "cx-portal-shared-components"
+import { useTranslation } from "react-i18next";
+
+const Header =() => {
```
Comment (JaneMcBrain) [inline cx-portal/src/components/pages/DigitalTwins/Header.tsx:4]: @oyo I was wondering, why we are not having a component for this already? Because I just reused it from somewhere else. Registration Request and Network Partner is using the same code...

File: cx-portal/.env [hunk 2]
```diff
@@ -1,4 +1,5 @@
 REACT_APP_BASE_CENTRAL_IDP=https://catenaxdev003akssrv.germanywestcentral.cloudapp.azure.com/iamcentralidp/auth
 REACT_APP_BASE_UTIL=https://catenax-dev003-util.azurewebsites.net/v1
 REACT_APP_BPDM_API_BASE_URL=https://catenax-bpdm-dev.demo.catena-x.net/api
+REACT_APP_SLDT_API_BASE_URL=https://catenaxintakssrv.germanywestcentral.cloudapp.azure.com/semantics
```
Comment (oyo) [inline cx-portal/.env:4]: we just got rid of the .env file because this leads to images that are env dependent. Have a look at the new EnvironmentService that is supposed to return the Service URL depending on the document location. Some are still set to a fixed URL:
https://github.com/catenax-ng/product-portal-frontend/blob/main/cx-portal/src/services/EnvironmentService.ts#L26-L32
you could copy that `getBpdmApiBase()` function

File: cx-portal/src/state/features/digitalTwins/api.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import { HttpClient } from 'utils/HttpClient'
+import { FilterParams, ShellDescriptor, TwinList } from './types'
+import qs from 'querystring'
+
+export class DigitalTwinApi extends HttpClient {
+  private static classInstance?: DigitalTwinApi;
+
+
+  public constructor() {
+    super(`${process.env.REACT_APP_SLDT_API_BASE_URL}/twin-registry/registry/shell-descriptors`)
```
Comment (oyo) [inline cx-portal/src/state/features/digitalTwins/api.ts:11]: see my comment at the .env file. I suggest to replace this env var with `getSldtApiBase()`

File: cx-portal-shared-components/src/components/basic/CustomAccordion/CustomAccordion.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    <Typography>Hello world</Typography>
+  </Component>
+)
+
+export const CustomAccordion = Template.bind({})
+CustomAccordion.args = {
+  expanded: true,
+  id: 'panel-1',
+  title: 'Accordion Title'
+}
```
Comment (oyo) [inline cx-portal-shared-components/src/components/basic/CustomAccordion/CustomAccordion.stories.tsx:24]: I found it hard to follow how the accordion works in the storybook - can we somehow show an example with multiple accordions?

**Review by oyo — CHANGES_REQUESTED** [review]
nicely integrated but some minor points from my side mostly regarding the accordion and the env vars. maybe we can sync this week to talk about that.

File: cx-portal-shared-components/src/components/basic/CustomAccordion/CustomAccordion.stories.tsx (hunk 4 above)
Comment (JaneMcBrain) [reply cx-portal-shared-components/src/components/basic/CustomAccordion/CustomAccordion.stories.tsx:24]: yes, at the moment I didn't implement it properly... will fix it.

File: .husky/pre-commit [hunk 5]
```diff
@@ -1,5 +1,5 @@
 #!/bin/sh
 . "$(dirname "$0")/_/husky.sh"
 
-npm run pretty
-npm run lint --fix
+# npm run pretty
+# npm run lint --fix
```
[the remark this answers is not in the document]
Comment (JaneMcBrain) [reply .husky/pre-commit:5]: When running yarn pretty I get 84 file changes - I am not sure, if this is intended - can I run yarn pretty for specific files instead?

**Comment by oyo** [comment]
@JaneMcBrain looks quite good now - do you also want to have a look at those two code smells from the Sonarcloud scan? I guess both should be easy to fix.

**Review by oyo — APPROVED** [review]
passt
